vehicle/src/Vehicle/Data/Builtin/Standard/Instances.hs

```diff
@@ -177,7 +177,21 @@ allInstances =
                 tUnit,
             False
           ),
-          ----------------
+          -------------------------
+          -- ValidTensorLikeType --
+          -------------------------
+
+          ( forAllTypes $ \t ->
+              forAllDims $ \ds ->
+                isTensorType t ds
```

Comment:
I don't think we actually need this `isTensorType t ds` constraint? This will get generated and checked when the user writes `Tensor X Y`.